components/notebook-controller/controllers/notebook_controller.go

```diff
@@ -78,6 +83,24 @@ func (r *NotebookReconciler) Reconcile(req ctrl.Request) (ctrl.Result, error) {
 	ctx := context.Background()
 	log := r.Log.WithValues("notebook", req.NamespacedName)
 
+	event := &v1.Event{}
+	var getEventErr error
+	getEventErr = r.Get(ctx, req.NamespacedName, event)
+	if getEventErr == nil {
+		involvedNotebook := &v1beta1.Notebook{}
+		involvedNotebookKey := types.NamespacedName{Name: nbNameFromEvent(event), Namespace: req.Namespace}
+		if err := r.Get(ctx, involvedNotebookKey, involvedNotebook); err != nil {
```

Comment:
Correct me if I am wrong, but it seems to be that it will watch all events from statefulset and pods. So we will get events from unrelated ss/pods?

Reply:
Seems not ideal if this will be triggered by unrelated events.
WDYT @jlewi ?
Can we filter using event.involvedObject.name?